src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -877,28 +881,38 @@ private static void ThrowOOM()
                 throw CreateHttpRequestException(new CurlException((int)CURLcode.CURLE_OUT_OF_MEMORY, isMulti: false));
             }
 
-            internal sealed class SendTransferState
+            internal sealed class SendTransferState : IDisposable
```

Comment:
That would add cost and shouldn't be necessary.  When a request is made, we have to keep the state associated with the request's processing rooted until a response object is published to the caller, as until then there's nothing for then to hold on to other than the ```Task<HttpResponseMessage>``` and the native side of things will be calling back and expecting to find its state.  Then once the response is published, that response stream has a finalizer, so whether that object is explicitly Disposed or not, all of this state will eventually be cleaned up.